### PR TITLE
Feature/db performance audit

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- [x] Make sure that you used the correct Gitflow.
- [x] Make sure that you documented your work in a professional way.
### Project requirements
- [x] EXPLAIN ANALYZE used
- [x] Add an extra column to the owners table
- [x] Statements to add data to the database
- [x] Find a way to decrease the execution time of the first query
- [x] EXPLAIN ANALYZE results (before and after improving the speed) for each of the 3 problematic queries.
- [x] EXPLAIN ANALYZE results before
explain1
explain2
explain3
- [x] Create an Index to optimize
create index
- [x] EXPLAIN ANALYZE results After
explain1a
explain2a
explain3a

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -228,3 +228,14 @@ INSERT INTO visits(animal_id, vet_id, date_of_visit) VALUES
...
+-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
+INSERT INTO visits (animals_id, vets_id, date_of_visit) 
+SELECT * FROM (SELECT id FROM animals) animal_ids, 
+(SELECT id FROM vets) vets_ids, 
+generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') 
+visit_timestamp;
+
+-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
+Insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000),
+ 'owner_' || generate_series(1,2500000) || '@mail.com';
```
Comment (Gambit142) [inline data.sql:241]: - [ ] Great work writing these `INSERT` statements. After populating the necessary tables with these data, the outcome of each of the following queries written below should have an execution time that is **_greater or equal_** to `1000ms`. 

- ` EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;`
- `EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;`
- `EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';`

I would advise you to run the statements that insert data into the table two or three more times, then run the `explain analyze` statement and take screenshots of the execution time that are equal to or more than _**1 second (1000ms)**_. 👍 😎 

Kindly attach the screenshots of the queries to your pull request review showing with each screenshot display an execution time of _**>=1000**_. After these, you then rerun the SQL statements that apply indexes to the special columns and then run each of the queries above again. 😎 

Thanks to the indexes applied, their execution time should be **_significantly lower_** than `1000ms`. Take the screenshots and attach them to your pull request summary. I understand that the current screenshots after running the indexes are significantly low, but you would agree that if you insert more data into the tables, the execution time will be different even though you applied the indexes.

Kindly implement these changes. 👍 😎 

#### Hint:

- Run the queries before applying the indexes. Take screenshots of **_execution time above 1000ms_**
- Apply the indexes
- Run the queries again. Take screenshots of execution time **_significantly below 1000ms_**.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Benawi @wolterbwambale 👋, 

Good job so far!

## Highlights
- Populated database with data ✔️ 
- Created index on required columns ✔️ 
- Use of the `explain analyze` command ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: data.sql (hunk 1 above)
Comment (Benawi) [reply data.sql:241]: ## Changes ♻️
- [x] The statements that insert data into the table executed three more times
- [x] Execution time above 1000ms
explain1b
explain2b
explain3b
- [x] Execution time significantly below 1000ms.
explain1ba
explain2ba
explain3ba

**Review by BregornOriginal — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it 

Congratulations! 

allfather-approves-approved

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Good work implementing indexes

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**